lyse/__init__.py

```diff
@@ -422,6 +423,11 @@ def __init__(self, h5_path, run_paths, no_write=False):
             run_paths = run_paths['filepath']
         self.h5_path = h5_path
         self.no_write = no_write
+        try:
+            h5py.File(h5_path, 'r')
+        except:
```

Comment:
It is best to catch a specific exception class here. Could you catch the specific exception raised? (Presumably it is `FileNotFoundError`?)